### ISSUE TITLE
Check pkgid out of pkg file 

### ISSUE THREAD

**Issue by iradization** [autopkg/autopkg#754]
Hi, 

Currently, in order to check the pkgid attribute of a pkg file, i need to unpack it to some temp location 
```
pkgutil --expand <MY_PKG_FILE> <TEMP_FOLDER> 
```
than, reading from file `/<TEMP_FOLDER>/<MY_PKG_FILE>/PackageInfo` the first line and extract the field "identifier" out of it.

```
<pkg-info format-version="2" identifier="com.mypkg.id" version="1.0" relocatable="false" overwrite-permissions="false" followSymLinks="false" install-location="/" auth="root">
```

I was wondering, if there's any automated and robust way to do so, preferably using python only code, and without extracting the pkg contents to the disk. 

Basically my goal is to be able to check if a package is already installed by looking up the package file's package id in the list of package ids generated by `pkgutil --pkgs` 

Thanks !

**Comment by arubdesu** [comment]
If you're making a feature request for that to be possible with a particular processor in core that's one question, but you can certainly write a custom one that does so with the output path from FlatPkgUnpacker as an input. That being said, before worrying if python can do it, you'd want to ask yourself how you'd access it from the command line/zsh/bash to determine how you'd do it programmatically in python (would be my advice).

**Comment by gregneagle** [comment]
You can use `xar` to selectively extract files from a flat package as done here:
https://github.com/munki/munki/blob/main/code/client/munkilib/pkgutils.py#L410

"Basically my goal is to be able to check if a package is already installed by looking up the package file's package id in the list of package ids generated by pkgutil --pkgs"

Not really sure how that task is connected to autopkg...

**Comment by gregneagle** [comment]
Is this actually an autopkg issue, or are you just looking for help with a task you hope to use Python for?

**Comment by iradization** [comment]
Hi @gregneagle and @arubdesu and thanks for your insights, your help is much appreciated.

**Comment by macmule** [comment]
FWIW, we might have something of use: https://github.com/autopkg/dataJAR-recipes/blob/master/Shared%20Processors/DistributionPkgInfo.py
